### PR TITLE
Ft authorized routes edit events

### PR DESCRIPTION
**Pull request by meshnesh**
# What does this PR do?

This PR allows the owner of an event to be able to edit his events

## How should this be manually tested?

Clone the repo.
Install the packages npm install then run npm run dev to run the app.
Create an account on the app by registering, confirm your email address.
Add an event, navigate to manage events.
Click edit on the current event to edit it.

# Screenshots

screen shot 2018-06-04 at 12 08 30

### CODE REVIEW

File: src/components/Users/UserProfile.js [hunk 1]
```diff
@@ -7,32 +7,70 @@ import Navigation from '../Navigation/Navigation';
...
+			toogle:false,
+			selectedId: id,
+			eventDeleted:true
+		});
+		setTimeout(
+			function() {
+				this.props.history.push('/'); 
+			}
+				.bind(this), 1000);
+	}
```
[the remark this answers is not in the document]
Comment (meshnesh) [reply src/components/Users/UserProfile.js]: Thanks for the feedback I'll look into implementing it.

File: src/components/category/GetCategories.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		categories: state.authEvent.categories
+	};
+};
+
+const mapDispathToProps = (dispatch) => {
+	return {
+		getEventCategories: () => dispatch(getEventCategories())
+	};
+};
+export default connect(mapStateToProps, mapDispathToProps)(Category);
```
[the remark this answers is not in the document]
Comment (meshnesh) [reply src/components/category/GetCategories.js]: didn't even see this, thanks for noting it.

File: src/components/AuthEvents/EditEvent.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	componentWillReceiveProps(nextProps){
+		this.setState({eventData: nextProps.editEvent.event})
+	}
+
+    componentWillMount = () => {
+		let eventId = this.props.match.params.id
+        this.props.getEventAction(eventId)
+    }
+
+    handleInputChange(e) {
```
Comment (PaulKariukiRimiru) [inline src/components/AuthEvents/EditEvent.js]: what do you think about using arrow functions.

File: src/components/AuthEvents/EditEvent.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		const event = this.state.eventData;
+		let eventId = this.props.match.params.id
+		console.log('handleSubmit', this.props);
+		this.props.editEventAction(eventId, event);
+		this.setState({eventData: initialState});
+		setTimeout(
+			function() {
+				this.props.history.push('/profile');
+			}
+				.bind(this), 1000);
```
Comment (PaulKariukiRimiru) [inline src/components/AuthEvents/EditEvent.js]: Binding can be avoided by using arrow functions.

File: src/components/AuthEvents/EditEvent.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	render() {
+
+		const { eventData } = this.state
+		console.log(this.state)
```
Comment (PaulKariukiRimiru) [inline src/components/AuthEvents/EditEvent.js]: Ensure that you remove the `console.log()` statements

File: src/components/Users/UserProfile.js [hunk 6]
```diff
@@ -7,32 +7,70 @@ import Navigation from '../Navigation/Navigation';
...
+
+	constructor(props){
+		super(props);
+		this.state = {
+			toggle:false,
+			selectedId: null,
+			eventDeleted:false
+		};
+		this.toggleModal = this.toggleModal.bind(this);
+		this.onDelete = this.onDelete.bind(this);
```
Comment (PaulKariukiRimiru) [inline src/components/Users/UserProfile.js]: You can avoid binding your functions by using arrow functions

File: src/components/AuthEvents/EditEvent.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+
+class EditEvent extends Component{
+
+    constructor(props){
+		super(props);
+        this.state = {
+			eventData: initialState,
+		};
+		// console.log('Constructor',editEvent);
```
Comment (andela-jkanyua) [inline src/components/AuthEvents/EditEvent.js]: Remove  commented out code here

File: src/components/AuthEvents/EditEvent.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+
+	componentWillReceiveProps(nextProps){
+		this.setState({eventData: nextProps.editEvent.event})
+	}
+
+    componentWillMount = () => {
+		let eventId = this.props.match.params.id
```
Comment (andela-jkanyua) [inline src/components/AuthEvents/EditEvent.js]: Indentation Issue here
Also, check you linter rules as some places you are using semis others, you are not

File: src/components/AuthEvents/EditEvent.js [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	handleCategoryChange(event, index, value) {
+		const { eventData } = this.state
+		eventData.event_category = value
+		this.setState({eventData})
+	}
+
+	setFechaDesde(x,event){
+        // console.log('initial, ',event);
```
Comment (andela-jkanyua) [inline src/components/AuthEvents/EditEvent.js]: More commented code here

File: src/components/AuthEvents/EditEvent.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+								<label class="active" for="location">Event Location</label>
+							</div>
+							<div className="input-field col s6">
+								<input id="image_url" type="text" name="image_url" className="validate" value={ eventData.image_url } onChange={ this.handleInputChange } />
+								<label class="active" for="image_url">Image url</label>
+							</div>
+						</div>
+						<div className="row">
+							<div className="input-field col s4">
+								{/* <input id="date" type="text" name="date"className="validate" value={ eventData.date } onChange={ this.handleInputChange } />
```
Comment (andela-jkanyua) [inline src/components/AuthEvents/EditEvent.js]: Commented out code

File: src/reducers/initialState.js [hunk 11]
```diff
@@ -2,7 +2,7 @@ export default {
 	authLogin: {
 		authenticated: false,
 		loading: false,
-		login: true,
+		isLoggedIn: true,
```
Comment (andela-jkanyua) [inline src/reducers/initialState.js]: Should the default logged in state be true here?

File: README.md [hunk 12]
```diff
@@ -13,7 +13,7 @@ P.S: Feedback is welcome
 ### Install the dependacies
 
 * Run the command `npm install`
-* Run the project run the command `npm start`
+* Run the project run the command `npm run de`
```
Comment (andela-jkanyua) [inline README.md]: Should be `npm run dev` right?

File: src/components/AuthEvents/EditEvent.js (hunk 4 above)
Comment (meshnesh) [reply src/components/AuthEvents/EditEvent.js]: thank you, I'll take note and update accordingly

File: README.md (hunk 12 above)
Comment (meshnesh) [reply README.md]: yes, thanks for noting it.

File: src/actions/authEvents.actions.js [hunk 13]
```diff
@@ -17,6 +17,34 @@ export function createEventSuccess(payload) {
...
+}
+
+export function rsvpEventSuccess(payload) {
+	return {
+		type: types.SUCCESSFUL_RSVP,
+		payload
+	};
+}
+
 export function userEventsAction() {
```
Comment (Rodgers-M) [inline src/actions/authEvents.actions.js]: wouldn't it be better if you renamed this action in a way that explains what it does? so something like `getUserEventsAction`

File: src/components/AuthEvents/EditEvent.js [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+
+class EditEvent extends Component{
+
+    constructor(props){
+		super(props);
+        this.state = {
+			eventData: initialState,
+		};
+		this.handleInputChange = this.handleInputChange.bind(this);
```
Comment (Rodgers-M) [inline src/components/AuthEvents/EditEvent.js]: @meshnesh  as @PaulKariukiRimiru has mentioned below, using arrow functions saves you the hustle of having to bind `this` in every method

File: src/actions/authEvents.actions.js [hunk 15]
```diff
@@ -48,3 +76,62 @@ export function userCreateEventsAction(event) {
...
+			.then(resp => {
+				dispatch(editEventSuccess(resp.data));
+			})
+			.catch(err => {
+				dispatch(ajaxCallError(err));
+			});
+	};
+}
+
+export function deleteEventAction(eventId) {
```
Comment (Rodgers-M) [inline src/actions/authEvents.actions.js]: good flow of action dispatch

File: src/components/Users/UserProfile.js [hunk 16]
```diff
@@ -7,32 +7,70 @@ import Navigation from '../Navigation/Navigation';
...
 	}
 
-	
+	toggleModal(id) {
+		this.setState({ toggle: !this.state.toggle, selectedId: id});
+	}
+
+	onDelete(id) {
+		let eventId = this.state.selectedId;
+		this.props.deleteEventAction(eventId);
```
Comment (Rodgers-M) [inline src/components/Users/UserProfile.js]: What happens to the local state if an error occurs during event deletion? you could set state in a `.then()`, that way you would be sure it's only set if the operation was successful.

File: src/components/Users/UserProfile.js [hunk 17]
```diff
@@ -7,32 +7,70 @@ import Navigation from '../Navigation/Navigation';
...
+		this.setState({
+			toogle:false,
+			selectedId: id,
+			eventDeleted:true
+		});
```
Comment (Rodgers-M) [inline src/components/Users/UserProfile.js]: #L35-L40 What happens to the local state if an error occurs during event deletion? you could set state in a `.then()`, that way you would be sure it's only set if the operation was successful.

**Pull request merged**